### ISSUE TITLE
Nuxt vue/no-dupe-keys disallows TypeScript and default params support

### ISSUE THREAD

**Issue by daniluk4000** [vuejs/eslint-plugin-vue#1687]
**What rule do you want to change?**
vue/no-dupe-keys behavior in 8.0.0

**Does this change cause the rule to produce more or fewer warnings?**

1. When using TypeScript, asyncData values must be duplicated in data in order to make them typed
2. I may not return all params in asyncData, in that case data property will work like default value. 

**How will the change be implemented? (New option, new default behavior, etc.)?**
Rollback asyncData support for vue/no-dupe-keys

**Please provide some example code that this change will affect:**

```typescript
export default Vue.extend({
    async asyncData() {
        return {
            isPopupVisible: true,
        };
    },
    data() {
        return {
            isPopupVisible: false,
        };
    },
    mounted() {
        this.isPopupVisible = true;
        //...
     }
});
```

**What does the rule currently do for this code?**
`Duplicated key 'isPopupVisible'.(vue/no-dupe-keys)` for data property. If I remove it from data, I will get `Property 'isPopupVisible' does not exist on type 'CombinedVueInstance   >>'`

**What will the rule do after it's changed?**
This code works as intended.

**Additional context**
![изображение](https://user-images.githubusercontent.com/10450717/139423912-6aa747e0-cd6f-4c13-8dd1-cdea905ac899.png)

![изображение](https://user-images.githubusercontent.com/10450717/139423947-066f5fd4-2ca7-41f6-b136-916becbdfdfa.png)

New behavior leaves no choice but to disable this rule or place eslint-ignore everywhere. Yet, I have to stay on 7.x.

**Comment by daniluk4000** [comment]
As I've looked at vue/no-dupe-key documentation, I believe it is possible you simple allow us to exclude asyncData from inspection.

**Comment by FloEdelmann** [comment]
Indeed, `asyncData` is merged with `data`:

* https://nuxtjs.org/docs/directory-structure/pages#asyncdata
* https://github.com/nuxt/nuxt.js/blob/0742f59e2478bd3998fd4d49b6200584ee5e2721/packages/vue-app/template/utils.js#L88
